### PR TITLE
[1/10] CMSIS-NN graph partitioner for softmax

### PR DESCRIPTION
**Pull request by ashutosh-arm**
This is the first patch towards enabling use of CMSIS-NN in TVM. It introduces graph partitioner for CMSIS-NN. Please refer to https://github.com/apache/tvm-rfcs/pull/15.

cc: @areusch @Mousius @manupa-arm @mbaret

### CODE REVIEW

**Review by leandron — APPROVED** [review]
LGTM

**Review by Mousius — APPROVED** [review]

**Comment by leandron** [comment]
cc @areusch @mehrdadh @tqchen for reviews if possible

File: python/tvm/relay/op/contrib/cmsisnn.py
Comment (comaniac) [inline python/tvm/relay/op/contrib/cmsisnn.py:40]: The type should be Module as well

**Review by comaniac — APPROVED** [review]
LGTM. We should merge this after merging the RFC PR?

File: python/tvm/relay/op/contrib/cmsisnn.py
Comment (ashutosh-arm) [reply python/tvm/relay/op/contrib/cmsisnn.py:40]: Thanks Cody. I have pushed another commit.

**Comment by Mousius** [comment]
@comaniac do you think we could merge this before the RFC? It doesn't seem controversial in itself, and it'd be good to make progress on further changes whilst the RFC is being finalised

**Comment by comaniac** [comment]
I don't prefer to do so, but I'm open if others don't think this is an issue.

cc @areusch

**Comment by leandron** [comment]
The corresponding RFC apache/tvm-rfcs#15 is now merged, CI here is green, so I'm merging this PR.

Thanks @ashutosh-arm, @comaniac and @Mousius!

**Pull request merged**
